server/svix-server/src/v1/endpoints/endpoint.rs

```diff
@@ -691,4 +711,366 @@ mod tests {
+        let app_2 = create_test_app(&client, APP_NAME_2).await.unwrap();
+
+        let mut ep_1 = endpoint_in(EP_URI_APP_1_EP_1);
+        ep_1.uid = Some(uid.clone());
+
+        let mut ep_2 = endpoint_in(EP_URI_APP_2);
+        ep_2.uid = Some(uid.clone());
+
+        let ep_1 = post_endpoint_in(&client, &app_1, ep_1).await.unwrap();
+        let ep_2 = post_endpoint_in(&client, &app_2, ep_2).await.unwrap();
+
+        delete_endpoint(&client, &app_1, &ep_1.id).await.unwrap();
+        delete_endpoint(&client, &app_2, &ep_2.id).await.unwrap();
+
+        delete_test_app(&client, app_1).await.unwrap();
+        delete_test_app(&client, app_2).await.unwrap();
+    }
+
+    // Simply tests that upon rotating an endpoint secret that it differs from the prior one
+    #[tokio::test]
+    #[cfg_attr(not(feature = "integration_testing"), ignore)]
+    async fn test_endpoint_secret_get_and_rotation() {
+        let (client, _jh) = start_svix_server();
+
+        const APP_NAME: &str = "v1EndpointSecretRotationTestApp";
+        const EP_URI: &str = "http://v1EndpointSecretRotationTestEp.test";
+
+        let app_id = create_test_app(&client, APP_NAME).await.unwrap();
+
+        let ep = post_endpoint_default(&client, &app_id, EP_URI)
+            .await
+            .unwrap();
+
+        let former_secret: EndpointSecretOut = client
+            .get(
+                &format!("api/v1/app/{}/endpoint/{}/secret/", app_id, ep.id),
+                StatusCode::OK,
+            )
+            .await
+            .unwrap();
+
+        let _: EmptyResponse = client
+            .post(
+                &format!("api/v1/app/{}/endpoint/{}/secret/rotate/", app_id, ep.id),
+                serde_json::json!({ "key": null }),
+                StatusCode::NO_CONTENT,
+            )
+            .await
+            .unwrap();
+
+        assert!(
+            former_secret
+                != client
+                    .get(
+                        &format!("api/v1/app/{}/endpoint/{}/secret/", app_id, ep.id),
+                        StatusCode::OK
+                    )
+                    .await
+                    .unwrap()
+        );
+
+        let _: EmptyResponse = client
+            .post(
+                &format!("api/v1/app/{}/endpoint/{}/secret/rotate/", app_id, ep.id),
+                &former_secret,
+                StatusCode::NO_CONTENT,
+            )
+            .await
+            .unwrap();
+
+        assert_eq!(
+            former_secret,
+            client
+                .get(
+                    &format!("api/v1/app/{}/endpoint/{}/secret/", app_id, ep.id),
+                    StatusCode::OK
+                )
+                .await
+                .unwrap()
+        );
+    }
+
+    #[test]
+    #[cfg_attr(not(feature = "integration_testing"), ignore)]
+    fn test_endpoint_headers_crud() {
+        // TODO
+    }
+
+    // TODO: "Recover failed webhooks" after messages testing
```

Comment:
Fix the above comments, remove this, and let's get this PR in. We can then do another PR that puts this one in.